apps/web/lib/api/projects.ts

```diff
@@ -73,6 +75,18 @@ export const getProjectLayers = async (id: string) => {
   }
 };
 
+export const getProjectLayerById = async (projectId: string, id: string) => {
+  try {
+    const data = await getProjectLayers(projectId)
+    const temporaryLayer = data ? data.filter(layer => layer.id === id )[0] : undefined;
+    return temporaryLayer;
+  }
+  catch (error) {
+    console.error(error);
+    throw Error(`error: make sure you are connected to an internet connection!`)
+  }
+}
+
```

Comment:
I would avoid this here. The api scope should only be CRUD. The rest of the logic such as filtering etc should be elsewhere (component etc).